src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3063,6 +3113,19 @@ MillisecondsBetweenTimestamps(instr_time startTime, instr_time endTime)
 }
 
 
+/*
+ * MicrosecondsBetweenTimestamps is a helper to get the number of microseconds
+ * between timestamps when it is expected to be small enough to fit in a
+ * long.
```

Comment:
> when it is expected to be small enough to fit in a long.

If we use `uint64` as the return time like I suggested below, then this is 500k years. If we do that change, I don't think it's necessary to say that we expect it to fit in that.

If we keep a long, I think it's probably good to qualify how  long is "small enough to fit in a long". This is ~6 years.